greatestPalindrome.js

```diff
@@ -0,0 +1,56 @@
+
+/*Given a string return the longest palindrome that can be constructed by removing or shuffling characters. 
+Example: 
+'aha' -> 'aha' 
+'ttaatta' -> ' ttaaatt' 
+'abc' -> 'a' or 'b' or 'c' 
+'gggaaa' -> 'gaaag' or 'aggga' 
+Note if there are multiple correct answers you only need to return 1 palindrome.
+*/
+var word = "aabczRc";
+var wordDict = {};
+var leftPalindrome = [];
+var rigthPalindrome = [];
+var palindrome;
+
+function dictionaryCreator(word){
+  for(var index in word){
+    var letter = word[index];
+    if(wordDict[letter]){
+      wordDict[letter]++;
+    }
+    else{
+      wordDict[letter] = 1;
+    }
+  }
+};
+
+dictionaryCreator(word)
+console.log(wordDict)
+palindromeConstructor(wordDict)
+console.log(wordDict)
+console.log(palindrome)
+console.log(rigthPalindrome)
+
+
+function palindromeConstructor (wordDict){
```

Comment:
You can improve this by a lot by making  a better use of the dictionary you created, hence.
you know a palidrome should say the same left to right.
you only need to make a single loop.
if the dictionary item is even, put half of it on the right and half on the left,
if its uneven separate 1 into a singlets array or just one variable. then the even amount just separat as the previous step. at the end just join it with a middle value of the singles taken from odd dict items